src/pages/transactions/common/hooks/useTransactionColumns.tsx

```diff
@@ -0,0 +1,119 @@
+/**
+ * Invoice Ninja (https://invoiceninja.com).
+ *
+ * @link https://github.com/invoiceninja/invoiceninja source repository
+ *
+ * @copyright Copyright (c) 2022. Invoice Ninja LLC (https://invoiceninja.com)
+ *
+ * @license https://www.elastic.co/licensing/elastic-license
+ */
+
+import { Link } from '@invoiceninja/forms';
+import { transactionStatuses } from 'common/constants/transactions';
+import { route } from 'common/helpers/route';
+import { useFormatMoney } from 'common/hooks/money/useFormatMoney';
+import { useCountries } from 'common/hooks/useCountries';
+import { useCurrencies } from 'common/hooks/useCurrencies';
+import { Transaction } from 'common/interfaces/transactions';
+import { Badge } from 'components/Badge';
+import { DataTableColumns } from 'components/DataTable';
+import { useTranslation } from 'react-i18next';
+
+export const useTransactionColumns = () => {
+  const { t } = useTranslation();
+  const countries = useCountries();
+  const currencies = useCurrencies();
+  const formatMoney = useFormatMoney();
+
+  const getCurrencyCodeById = (currencyId: string): string => {
+    return currencies?.find(({ id }) => currencyId === id)?.code || '';
+  };
+
+  const getCountryIdByCurrencyCode = (
+    currencyCode: string | undefined
+  ): string => {
+    return (
+      countries?.find(({ currency_code: code }) => code === currencyCode)?.id ||
+      ''
+    );
+  };
+
```

Comment:
> This can be replaced by `useResolveCountry` hook.

We discussed about this case, and currently I kept it as is, but I will discuss with David about better solution.